src/rust/generators/sysmon-generator/src/serialization.rs

```diff
@@ -26,70 +26,66 @@ impl CheckedError for SysmonDecoderError {
             Self::DeserializeError(_) => Recoverable::Persistent,
             Self::DecompressionError(_) => Recoverable::Persistent,
             Self::TimeError(_) => Recoverable::Persistent,
+            Self::Utf8Error(_) => Recoverable::Persistent,
         }
     }
 }
 
 #[derive(Debug, Clone, Default)]
 pub struct SysmonDecoder;
 
-impl PayloadDecoder<Vec<Event>> for SysmonDecoder {
+impl<'a> PayloadDecoder<Vec<SysmonEvent<'static>>> for SysmonDecoder {
     type DecoderError = SysmonDecoderError;
 
     #[tracing::instrument(skip(self), err)]
-    fn decode(&mut self, body: Vec<u8>) -> Result<Vec<Event>, Self::DecoderError> {
+    fn decode(&mut self, body: Vec<u8>) -> Result<Vec<SysmonEvent<'static>>, Self::DecoderError> {
         let decompressed = grapl_service::decoder::decompress::maybe_decompress(body.as_slice())?;
 
         let mut first_deserialization_error: Option<SysmonDecoderError> = None;
 
-        /*
-           This iterator is taking a set of bytes of the logs, splitting the logs on newlines,
-           converting the byte sequences to utf-8 strings, and then filtering on supported event
-           types: Process Creation, Network Connection, and File Creation.
+        use sysmon_parser::system::EventId;
```

Comment:
even if a use is in a fn body it's global, so it should go at the top

Reply:
Sounds good.